shims/spark300/src/main/scala/com/nvidia/spark/rapids/shims/spark300/GpuHashJoin.scala

```diff
@@ -134,15 +135,17 @@ trait GpuHashJoin extends GpuExec with HashJoin {
       override def hasNext: Boolean = {
         while (nextCb.isEmpty && (first || stream.hasNext)) {
```

Comment:
As I mentioned in the previous PR, `stream.hasNext()` can be expensive.  Some iterators do their work in `hasNext()` rather than `next()` (i.e.: need to compute the next batch to see if there is a next batch, thus it may be attempting to grab the semaphore, fetch from disk, etc.).

We need to account for the time spent in `stream.hasNext()` or the time could be significantly underreported, similar to what was reported in #658.  The total time could be underreported in the same way.  I think we need to change where `startTime` is being setup to capture the time spent in `stream.hasNext()`.

Reply:
@jlowe  I have made a change to include the hasNext(); please review the code; Thanks.